### PR TITLE
Cleanup examples

### PR DESCRIPTION
**Pull request by adler-j**
A cleanup of the examples. They should all run now.

Changes

* BIG: changed old logic in `DiscreteLp.element`,  now tries calling `uspace.element` iff the input is a `callable` else tries the other one. This allows errors to trickle through properly. Much more understandable error messages.
* Removed `convolution_test_cuda` since it didnt realy fit.
* Removed `tomography` example since we have others now.
* Add a more odl'y tomography example under `solvers` using `conjugate_gradient_normal`.
* Fixed failing `DiscreteLp.examples` due to vectorization changes.

### CODE REVIEW

File: examples/solvers/conjugate_gradient_tomography.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Solves the optimization problem
+
+    min_x  1/2 ||A(x) - g||_2^2 + lam || |grad(x)| ||_1
+
+Where ``A`` is a parallel beam forward projector, ``grad`` the spatial
+gradient and ``g`` is given noisy data.
+
+For further details and a description of the solution method used, see
+:ref:`chambolle_pock` in the ODL documentation.
+"""
```
Comment (kohr-h) [inline examples/solvers/conjugate_gradient_tomography.py]: Old doc

File: odl/discr/lp_discr.py [hunk 2]
```diff
@@ -279,15 +279,12 @@ def element(self, inp=None, **kwargs):
         elif inp in self.dspace:
             return self.element_type(self, inp)
 
-        # uspace element -> discretize
-        try:
+        if callable(inp):
+            # uspace element -> discretize
```
Comment (kohr-h) [inline odl/discr/lp_discr.py]: More natural, agreed

File: examples/convolution_test_cuda.py [hunk 3]
```diff
@@ -1,96 +0,0 @@
...
-#
-# ODL is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
-
-"""Example of a deconvolution problem with different solvers (CUDA)."""
```
Comment (kohr-h) [inline examples/convolution_test_cuda.py]: Gone for good?

Comment (adler-j) [inline examples/convolution_test_cuda.py]: Yes. It could come back in some form with #336

**Comment by kohr-h** [comment]
Nice, most of all that there are more deletions than additions :-)

File: odl/discr/lp_discr.py (hunk 2 above)
Comment (adler-j) [inline odl/discr/lp_discr.py]: This change should make many things much more easy to deal with, especially error messages.

**Pull request merged**